hedera-node/test-clients/src/main/java/com/hedera/services/yahcli/commands/contract/evminfo/Assembly.java

```diff
@@ -0,0 +1,565 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.services.yahcli.commands.contract.evminfo;
+
+import com.hedera.services.yahcli.commands.contract.evminfo.DataPseudoOpLine.Kind;
+import com.hedera.services.yahcli.commands.contract.utils.Range;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HexFormat;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.IntFunction;
+import java.util.regex.Pattern;
+
+/**
+ * Generates disassembly of an EVM contract
+ *
+ * <p>Generates the disassembly of an EVM contract. Defines the _lines_ of a disassembly as falling
+ * into two classes: `Line`s - every line in the disassembly is one of these records, and `Code` -
+ * all bytes of the bytecode go into one of these records.
+ */
+public class Assembly {
+
+    /**
+     * A line of the disassembly
+     *
+     * <p>(dev note: here is an _interface_ with _default_ methods so that "subclasses" can be
+     * immutable records - which is what I really want)
+     */
+    public interface Line {
+
+        /** Formats the line into a string suitable for directly putting in the disassembly */
+        void formatLine(@NonNull final StringBuilder sb);
+
+        /**
+         * Formats the line into a string suitable for directly putting in the disassembly
+         *
+         * <p>Convenience method so that holders of Lines don't have to create their own
+         * StringBuilder
+         */
+        default @NonNull String formatLine() {
+            final var sb = new StringBuilder(MAX_EXPECTED_LINE);
+            formatLine(sb);
+            return sb.toString();
+        }
+
+        /**
+         * Extend a string until it reaches the goal column
+         *
+         * <p>Utility function useful for lining up fields (opcode, mnemonic, comment, etc.) into
+         * columns.
+         */
+        default void extendWithBlanksTo(@NonNull final StringBuilder sb, final int goalColumn) {
+            if (goalColumn <= sb.length()) {
+                // Already past the goal column ... just pad with a couple of blanks
+                sb.append("  ");
+                return;
+            }
+
+            // Need to pad to goal column - keep sticking blanks on to the end until past
+            // the goal, then truncate to correct size
+            while (sb.length() < goalColumn) {
+                sb.append("                ");
+            }
+            sb.setLength(goalColumn);
+        }
+
+        /**
+         * Check if this Line is a specific instruction opcodes
+         *
+         * <p>Convenience method for check to see if this line is an instruction with the given
+         * opcode.
+         */
+        default boolean thisLineIsA(final int opcode) {
```

Comment:
Why the overload of `thisLineIsA`?
Would this be better names `isCodeLineOpcodeMatch()` and `isCodeLineMneumonicMatch()`